profiles/models.py

```diff
@@ -87,7 +87,7 @@ class Profile(models.Model):
     )
 
     def save(self, *args, **kwargs):
-        if not self.pk and self.user:
+        if not (self.first_name and self.last_name) and not self.pk and self.user:
```

Comment:
this was in a nice own commit, now it is lumped with the changes it has nothing to do with 😅

Reply:
I would maybe start with checking, if the object has `pk` first, as it would probably already stop the logic for so many cases. then you could do a slightly "heavier" `not (self.first_name and self.last_name)`, and then check, if the object has the user set.

Reply:
```python
if not self.pk and not (self.first_name and self.last_name) and self.user:
    ...
```

Reply:
> this was in a nice own commit, now it is lumped with the changes it has nothing to do with 😅

This was a mistake by me when squashing the commits. My bad. I noticed it when I was writing the tests for the actual task here.